internal/api/hls.go

```diff
@@ -30,16 +30,25 @@ func (a *ApiManagerCtx) HLS(r chi.Router) {
 			return
 		}
 
+		_, stream_exists := a.Conf.Streams[input]
+		if !stream_exists {
+			w.WriteHeader(http.StatusNotFound)
+			w.Write([]byte("404 not found"))
+			return
+		}
+
 		ID := fmt.Sprintf("%s/%s", profile, input)
 
 		manager, ok := hlsManagers[ID]
 		if !ok {
 			// create new manager
 			manager = hls.New(func() *exec.Cmd {
 				// get transcode cmd
-				cmd, err := transcodeStart("profiles/hls", profile, input)
+				cmd, err := a.transcodeStart("hls", profile, input)
 				if err != nil {
-					logger.Panic().Err(err).Msg("transcode could not be started")
+					logger.Error().Err(err).Msg("transcode could not be started")
+					w.WriteHeader(http.StatusInternalServerError)
+					w.Write([]byte(fmt.Sprintf("%v\n", err)))
```

Comment:
We cannot return HTTP error in callback function, this leads to panic, since at that point the request must not exist.

![image](https://user-images.githubusercontent.com/7534274/134781764-40396a1e-24be-4b90-8b43-2e81d4d252c2.png)

Reply:
good point! i think you fixed it

Reply:
It works if source is not found, but not when transcode profile does not exist. So it needs still a fix.

Reply:
ok i think i fix it now